specification/synonym_rule/put/SynonymRulePutResponse.ts

```diff
@@ -0,0 +1,24 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { SynonymsUpdateResult } from '../../synonyms/_types/SynonymsUpdateResult'
+
+export class Response {
```

Comment:
There's some duplication between this response signature and others such as `SynonymRuleDeleteResponse`. Is this intentional or is there any way to consolidate?

Reply:
I tried to consolidate as much as possible by creating the `SynonymsUpdateResult` type, which is essentially the body for these responses.

I'd say we need the separate Request / Response classes for the codegen to work - hopefully the clients team can recommend a better pattern here.

Reply:
> I'd say we need the separate Request / Response classes for the codegen to work - hopefully the clients team can recommend a better pattern here.

Yes we need those for codegen, each request/response couple will be generated into specific expanded types with their namespace.
In this case it does seem like there's duplication but I don't have a better pattern, you went as far as we can at the moment.

Kudos for a consistent API! :+1: